src/Core/Scaffold/PayloadBuilder.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace GetCandy\Api\Core\Scaffold;
+
+use Illuminate\Support\Str;
+use Illuminate\Database\Eloquent\Model;
+use Schema;
+use Illuminate\Validation\Validator;
+use Illuminate\Contracts\Validation\Factory;
+
+class PayloadBuilder
+{
+    protected $payload;
+
+    public function __construct(PayloadInterface $payload = null)
+    {
+        $this->setPayload($payload);
+    }
+
+    public function validate(array $data, array $messages = [], array $customAttributes = [])
+    {
+        return $this->getValidationFactory()->make(
+            $data, $this->payload->rules, $messages, $customAttributes
+        )->validate();
+    }
+
+    public function prepare($request)
+    {
+        foreach ($request->all() as $key => $value) {
+            $method = "set" . Str::camel($key) . "Attribute";
+            if (method_exists($this->payload, $method)) {
+                $this->payload->{$method}($value);
+            } else {
+                $this->payload->setAttribute($key, $value);
+            }
+        }
+        return $this->payload;
+    }
+
+    public function setPayload(PayloadInterface $payload)
+    {
+        $this->payload = $payload;
+    }
+
+    public function validationRules()
+    {
+        return $this->payload->rules;
+    }
+
+    public function getModelAttributes(Model $model)
+    {
+        $columns = Schema::getColumnListing($model->getTable());
```

Comment:
Do we really need to check the model's fields? Surely we can trust the payload?

Reply:
I guess we have the validation, it was just an idea really, can revisit it

Reply:
I think it would clean things up a bit, I don't see a need for it.

Reply:
I guess we'd still need a way to sanitise the data coming in from the request 🤔